docs/keda-installation.md

```diff
@@ -0,0 +1,316 @@
+kubectl apply -f config/samples/operator_v1alpha1_keda_k3d.yaml
+```
+## Installation on Kyma runtime
+
+This section describes how to set up the Keda module (Keda + Keda Manager) on top of the Kyma installation with `lifecycle-manager`.
+In such a setup, you don't need to install Keda Manager. It is installed and managed by `lifecycle-manager`.
+
+### Lifecycle management of Keda Manager in Kyma
+
+When you enable the Keda module using your Kyma runtime Kyma custom resource (CR), the `lifecycle-manager` downloads the bundled package of the Keda Manager and installs it. Additionally, it applies a sample Keda CR, which triggers Keda Manager to install the Keda module.
+
+![a](./docs/assets/keda-lm-overview.drawio.svg)
+
+In order to be consumable by `lifecycle-manager`, the Keda module is bundled in a [specific way](https://github.com/kyma-project/community/tree/main/concepts/modularization#component-packaging-and-versioning) and described using the special ModuleTemplate CR.
+
+## Local k3d setup for the local lifecycle-manager & Keda Manager
+
+> **TIP:** Use the dedicated `make` target (in the `hack` folder) to execute the following steps automatically.
+
+1. Clone the project.
+
+   ```bash
+   git clone https://github.com/kyma-project/keda-manager.git && cd keda-manager/
+   ```
+
+2. Provide the k3d cluster.
+
+   ```bash
+   kyma provision k3d
+   ```
+
+3. Build and push the Keda Manager image.
+
+   ```bash
+   make module-image IMG_REGISTRY=localhost:5001/unsigned/manager-images IMG=localhost:5001/keda-manager-dev-local:0.0.2
+   ```
+
+4. Build and push the Keda module.
+
+   ```bash
+   make module-build IMG=k3d-kyma-registry:5001/keda-manager-dev-local:0.0.2 MODULE_REGISTRY=localhost:5001/unsigned
+   ```
+
+5. Verify if the module and the manager's image are pushed to the local registry.
+
+   ```bash
+   curl localhost:5001/v2/_catalog
+   ```
+You should get a result similar to this example:
+
+   ```json
+   {"repositories":["keda-manager-dev-local","unsigned/component-descriptors/kyma-project.io/module/keda"]}
+   ```
+6. Inspect the generated module template.
+
+> **NOTE:** The following sub-steps are temporary workarounds.
+
+Edit `template.yaml` and:
+- change `target` to `control-plane`
+
+   ```yaml
+   spec:
+    target: control-plane
+    ```
+> **NOTE:** This is required in the single-cluster mode only.
+
+- change the existing repository context in `spec.descriptor.component`:
+
+   ```yaml
+   repositoryContexts:      
+     - baseUrl: k3d-kyma-registry.localhost:5000/unsigned
+       componentNameMapping: urlPath
+       type: ociRegistry
+   ```
+
+7. Install the modular Kyma on the k3d cluster.
+
+> **NOTE** This installs the latest versions of `lifecycle-manager`.
+
+Use the `--template` flag to deploy the Keda module manifest from the beggining, or apply it using kubectl later.
+
+   ```bash
+   kyma alpha deploy --template=./template.yaml
+   ```
+Kyma installation is ready, but the module is not yet activated.
```

Comment:
```suggestion
   When the Kyma installation is ready, activate the module:
```

Reply:
Wait, do we only activate it in line 109? Then what do we do here? Make it known?